crates/firewheel-core/src/dsp/pan_law.rs

```diff
@@ -66,3 +71,32 @@ impl PanLaw {
         }
     }
 }
+
+impl Diff for PanLaw {
+    fn diff<E: crate::diff::EventQueue>(
+        &self,
+        baseline: &Self,
+        path: crate::diff::PathBuilder,
+        event_queue: &mut E,
+    ) {
+        if self != baseline {
+            event_queue.push_param(*self as u32, path);
+        }
+    }
+}
+
+impl Patch for PanLaw {
+    fn patch(
+        &mut self,
+        data: &crate::event::ParamData,
+        _path: &[u32],
+    ) -> Result<(), crate::diff::PatchError> {
+        match data {
+            ParamData::U32(raw) => {
+                *self = Self::from_u32(*raw);
+                Ok(())
+            }
+            _ => Err(crate::diff::PatchError::InvalidData),
+        }
+    }
+}
```

Comment:
We could easily implement derive macros for diffing and patching enums in follow-up work.